### PR TITLE
jv-stream-github-practice first solution

### PR DESCRIPTION
**Pull request by kcasPilot**

### CODE REVIEW

File: src/main/java/practice/StreamPractice.java [hunk 1]
```diff
@@ -14,7 +21,14 @@ public class StreamPractice {
      * "Can't get min value from list: < Here is our input 'numbers' >"
      */
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        return numbers.stream()
+                .map(s -> Arrays.stream(s.split(NUMBERS_SEPARATOR)))
+                .flatMap(Stream::sequential)
```
Comment (olekskov) [inline src/main/java/practice/StreamPractice.java:26]: can we use just flatMap with map previously?

File: src/main/java/practice/StreamPractice.java [hunk 2]
```diff
@@ -59,7 +85,13 @@ public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
      * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
      */
     public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        Predicate<Person> female = new PersonFilter(femaleAge, Person.Sex.WOMAN);
+        return peopleList.stream()
+                .filter(female)
+                .map(Person::getCats)
+                .flatMap(List::stream)
```
Comment (olekskov) [inline src/main/java/practice/StreamPractice.java:92]: you may flatmap without preceeding map here, think of argument

File: src/main/java/practice/StreamPractice.java [hunk 3]
```diff
@@ -14,7 +21,13 @@ public class StreamPractice {
      * "Can't get min value from list: < Here is our input 'numbers' >"
      */
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        return numbers.stream()
+                .flatMap(s -> Arrays.stream(s.split(NUMBERS_SEPARATOR)))
+                .mapToInt(Integer::parseInt)
+                .filter(i -> i % 2 == 0)
```
Comment (Ivan95kos) [inline src/main/java/practice/StreamPractice.java:27]: For the future, do not use short names that are difficult to read
```suggestion
                .flatMap(stringNumbers -> Arrays.stream(stringNumbers.split(NUMBERS_SEPARATOR)))
                .mapToInt(Integer::parseInt)
                .filter(number -> number % 2 == 0)
```

File: src/main/java/practice/StreamPractice.java [hunk 4]
```diff
@@ -23,7 +36,14 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        Function<Integer, Integer> oddIndexValueChanger = new OddIndexChanger();
+        return numbers.stream()
+                .mapToInt(oddIndexValueChanger::apply)
+                .filter(i -> i % 2 != 0)
```
Comment (Ivan95kos) [inline src/main/java/practice/StreamPractice.java:42]:
```suggestion
                .filter(number -> number % 2 != 0)
```

File: src/main/java/practice/StreamPractice.java [hunk 5]
```diff
@@ -59,7 +86,12 @@ public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
      * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
      */
     public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        Predicate<Person> female = new PersonFilter(femaleAge, Person.Sex.WOMAN);
+        return peopleList.stream()
+                .filter(female)
+                .flatMap(p -> p.getCats().stream())
```
Comment (Ivan95kos) [inline src/main/java/practice/StreamPractice.java:92]:
```suggestion
                .flatMap(person -> person.getCats().stream())
```

**Review by Ivan95kos — APPROVED** [review]
Great job, with minor comments

**Pull request closed without merging**